src/dsl/tokenizer.rs

```diff
@@ -83,6 +83,51 @@ impl<'input> Tokenizer<'input> {
 
         Some(Ok(self.cache.as_ref().unwrap()))
     }
+
+    pub fn can_get_idx(&mut self) -> Option<Result<bool, ParseError>> {
+        self.peek().map(|res| {
+            res.map(|token| match &token.kind {
+                TokenKind::Key(k) => k
+                    .chars()
+                    .next()
+                    .map(|c| c.is_ascii_digit())
+                    .unwrap_or(false),
+                _ => false,
+            })
+        })
+    }
+
+    pub fn get_idx(&mut self) -> usize {
+        let token = self.next().unwrap().unwrap();
```

Comment:
unwraps here can be replaced by error propagation

Reply:
idea with this is caller should never call this without calling `can_get_idx` which handles errors. I'll add comment

Reply:
https://github.com/infinyon/fluvio-jolt/pull/26#discussion_r1186032560

Reply:
This makes it brittle.   This should have a better internal contract.

Also, why does this need to be `mut`?  If so, change name to  to something like `get_mut_index`